In [248]:
import pandas as pd
import numpy as np


In [249]:
heroes = pd.read_json("./purchase_data2.json")
heroes_df = pd.DataFrame(heroes)
heroes_df.head()

Age Gender  Item ID                 Item Name  Price          SN
0   20   Male       93  Apocalyptic Battlescythe   4.49     Iloni35
1   21   Male       12                     Dawne   3.36   Aidaira26
2   17   Male        5                Putrid Fan   2.63      Irim47
3   17   Male      123         Twilight's Carver   2.55     Irith83
4   22   Male      154              Feral Katana   4.11  Philodil43

In [250]:
Players = heroes_df["SN"].unique()
num=len(Players)
number_Players = pd.DataFrame({"Total Players": [num]})
number_Players

Total Players
0             74

Number of unique items


In [251]:

Uniqe_Item=heroes_df["Item ID"].unique()
Unique_count=len(Uniqe_Item)
Average_Purchase_Price = heroes_df["Price"].mean()
Total_num_Purchase = heroes_df["Item ID"].count()
Total_Revenue = heroes_df["Price"].sum()
purchasing_analysis= pd.DataFrame({"Number of unique items":[Unique_count],
                                   "Average Purchase Price":[Average_Purchase_Price],
                                   "Number of Purchases":[Total_num_Purchase],
                                   "Total Revenue":[Total_Revenue]})
purchasing_analysis

Average Purchase Price  Number of Purchases  Number of unique items  \
0                2.924359                   78                      64   

   Total Revenue  
0          228.1

Gender Demographics


In [252]:

Female = len([Gender for Gender in heroes_df["Gender"] if Gender=="Female"])
Male = len([Gender for Gender in heroes_df["Gender"] if Gender=="Male"])
Non_disclosed = len([Gender for Gender in heroes_df["Gender"] if (Gender!="Female" and Gender != "Male" )])
Total = Male+Female+Non_disclosed
Gender_dem_df = pd.DataFrame ({"":["Female", "Male", "Other / Non-Disclosed"],
                              "Percentage of Players":[100*Female/Total, 100*Male/Total, 100*Non_disclosed/Total],
                              "Total Count":[Female, Male, Non_disclosed]})
Gender_d#Purchasing Analysis (Gender)
em_df


Percentage of Players  Total Count
0                 Female              16.666667           13
1                   Male              82.051282           64
2  Other / Non-Disclosed               1.282051            1

Purchasing Analysis (Gender)


In [253]:

a=heroes["Gender"].value_counts()
a=a.to_frame()
heroes_df
heroes_df_grouped = heroes_df[["Gender", "Price"]].groupby(["Gender"]).agg(["count","sum", "mean"])

heroes_df_grouped.columns = ["Purchase Count", "Total Purchase Value", "Average Purchase Value"]
heroes_df_grouped["Normalized Totals"] = heroes_df_grouped["Total Purchase Value"]/Gender_dem_df["Total Count"].sum()
heroes_df_grouped 


Purchase Count  Total Purchase Value  \
Gender                                                        
Female                             13                 41.38   
Male                               64                184.60   
Other / Non-Disclosed               1                  2.12   

                       Average Purchase Value  Normalized Totals  
Gender                                                            
Female                               3.183077           0.530513  
Male                                 2.884375           2.366667  
Other / Non-Disclosed                2.120000           0.027179

Age Demographics


In [254]:

bins = [0,10,14,19,24,29,34,39,100]
group_name = ["<10", "10-14","15-19", "20-24", "25-29","30-34", "35-39", "40+" ]
heroes_df["Age Range"] = pd.cut(heroes_df["Age"], bins, labels=group_name)
age_range_grouped = heroes_df[["Age Range","Gender"]].groupby(["Age Range"]).count()
Total_count = age_range_grouped["Gender"].sum()
age_range_grouped["Total Count"] = age_range_grouped["Gender"]*100/Total_count
age_range_grouped = age_range_grouped.rename(columns={"Gender":"Total Count", "Total Count":"Percentage of Players"})
age_range_grouped

Total Count  Percentage of Players
Age Range                                    
<10                  5               6.410256
10-14                3               3.846154
15-19               11              14.102564
20-24               36              46.153846
25-29                9              11.538462
30-34                7               8.974359
35-39                6               7.692308
40+                  1               1.282051

Average Purchase Price


In [255]:

heroes_grouped =heroes_df[["Age Range", "SN", "Price"]].groupby(["Age Range"]).agg({"Price" : ["count", "mean", "sum"]})
heroes_grouped.columns = ["Purchase Count", "Average Purchase Price", "Total Purchase Value"]
heroes_grouped["Normalized Totals"] = heroes_grouped["Total Purchase Value"]/heroes_grouped["Total Purchase Value"].sum()
heroes_grouped

Purchase Count  Average Purchase Price  Total Purchase Value  \
Age Range                                                                 
<10                     5                2.764000                 13.82   
10-14                   3                2.986667                  8.96   
15-19                  11                2.764545                 30.41   
20-24                  36                3.024722                108.89   
25-29                   9                2.901111                 26.11   
30-34                   7                1.984286                 13.89   
35-39                   6                3.561667                 21.37   
40+                     1                4.650000                  4.65   

           Normalized Totals  
Age Range                     
<10                 0.060587  
10-14               0.039281  
15-19               0.133319  
20-24               0.477378  
25-29               0.114467  
30-34               0.060894  
35-39               0.093687  
40+                 0.020386

Top Spenders


In [279]:

Top_Spenders = heroes_df[["SN", "Price"]].groupby(["SN"]).agg({"Price":["count","mean","sum"]})
Top_Spenders.columns=["Purchase Count", "Average Purchase Price", "Total Purchase Value"]
Top_Spenders.sort_values("Total Purchase Value" , ascending=False).head(5)

Purchase Count  Average Purchase Price  Total Purchase Value
SN                                                                      
Sundaky74                2                   3.705                  7.41
Aidaira26                2                   2.565                  5.13
Eusty71                  1                   4.810                  4.81
Chanirra64               1                   4.780                  4.78
Alarap40                 1                   4.710                  4.71

Most Popular Items


In [271]:

Most_Popilar_Item = heroes_df[["Item ID","Item Name","Price"]].groupby(["Item ID", "Item Name", "Price"]).agg({"Price":["count", "sum"]})
Most_Popilar_Item.columns=["Purchase Count", "Total Purchase Value"]
Most_Popilar_Item.sort_values("Purchase Count", ascending=False).head(5)

Purchase Count  Total Purchase Value
Item ID Item Name      Price                                      
94      Mourning Blade 3.64                3                 10.92
90      Betrayer       4.12                2                  8.24
111     Misery's End   1.79                2                  3.58
64      Fusion Pummel  2.42                2                  4.84
154     Feral Katana   4.11                2                  8.22

Most Profitable Items


In [272]:
Most_Popilar_Item.sort_values("Total Purchase Value", ascending=False).head(5)


Purchase Count  \
Item ID Item Name                         Price                   
94      Mourning Blade                    3.64                3   
117     Heartstriker, Legacy of the Light 4.71                2   
93      Apocalyptic Battlescythe          4.49                2   
90      Betrayer                          4.12                2   
154     Feral Katana                      4.11                2   

                                                 Total Purchase Value  
Item ID Item Name                         Price                        
94      Mourning Blade                    3.64                  10.92  
117     Heartstriker, Legacy of the Light 4.71                   9.42  
93      Apocalyptic Battlescythe          4.49                   8.98  
90      Betrayer                          4.12                   8.24  
154     Feral Katana                      4.11                   8.22

Report

- 82% of the players are male
- Most of the players are between 20-24 years old
- Average purchase price from females are higher than men
